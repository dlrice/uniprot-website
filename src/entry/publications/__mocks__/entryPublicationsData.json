{
  "facets": [
    {
      "label": "Scale",
      "name": "scale",
      "allowMultipleSelection": false,
      "values": [
        {
          "value": "Small",
          "count": 2272
        },
        {
          "value": "Large",
          "count": 116
        }
      ]
    }
  ],
  "results": [
    {
      "id": "1",
      "literatureEntry": {
        "pubmedId": 2881207,
        "doiId": "10.1038/325733a0",
        "title": "The precursor of Alzheimer's disease amyloid A4 protein resembles a cell- surface receptor.",
        "authors": [
          "Kang J.",
          "Lemaire H.-G.",
          "Unterbeck A.",
          "Salbaum J.M.",
          "Masters C.L.",
          "Grzeschik K.-H.",
          "Multhaup G.",
          "Beyreuther K.",
          "Mueller-Hill B."
        ],
        "completeAuthorList": true,
        "publicationDate": "1987",
        "journal": "Nature",
        "firstPage": "733",
        "lastPage": "736",
        "volume": "325",
        "literatureAbstract": "Alzheimer's disease is characterized by a widespread functional disturbance of the human brain. Fibrillar amyloid proteins are deposited inside neurons as neurofibrillary tangles and extracellularly as amyloid plaque cores and in blood vessels. The major protein subunit (A4) of the amyloid fibril of tangles, plaques and blood vessel deposits is an insoluble, highly aggregating small polypeptide of relative molecular mass 4,500. The same polypeptide is also deposited in the brains of aged individuals with trisomy 21 (Down's syndrome). We have argued previously that the A4 protein is of neuronal origin and is the cleavage product of a larger precursor protein. To identify this precursor, we have now isolated and sequenced an apparently full-length complementary DNA clone coding for the A4 polypeptide. The predicted precursor consists of 695 residues and contains features characteristic of glycosylated cell-surface receptors. This sequence, together with the localization of its gene on chromosome 21, suggests that the cerebral amyloid deposited in Alzheimer's disease and aged Down's syndrome is caused by aberrant catabolism of a cell- surface receptor.",
        "statistics": {
          "reviewedProteinCount": 1,
          "unreviewedProteinCount": 0,
          "mappedProteinCount": 0
        }
      },
      "uniProtReference": {
        "citation": {
          "citationType": "journal article",
          "authors": [
            "Kang J.",
            "Lemaire H.-G.",
            "Unterbeck A.",
            "Salbaum J.M.",
            "Masters C.L.",
            "Grzeschik K.-H.",
            "Multhaup G.",
            "Beyreuther K.",
            "Mueller-Hill B."
          ],
          "citationXrefs": [
            {
              "databaseType": "PubMed",
              "id": "2881207"
            },
            {
              "databaseType": "DOI",
              "id": "10.1038/325733a0"
            }
          ],
          "title": "The precursor of Alzheimer's disease amyloid A4 protein resembles a cell-surface receptor.",
          "publicationDate": "1987",
          "journal": "Nature",
          "firstPage": "733",
          "lastPage": "736",
          "volume": "325"
        },
        "referencePositions": ["NUCLEOTIDE SEQUENCE [MRNA] (ISOFORM APP695)"],
        "referenceComments": [
          {
            "value": "Brain",
            "type": "TISSUE"
          }
        ]
      },
      "categories": ["Sequence", "Sequence"],
      "publicationSource": "Swiss-Prot"
    },
    {
      "id": "2",
      "literatureEntry": {
        "pubmedId": 10097139,
        "doiId": "10.1073/pnas.96.7.3922",
        "title": "Constitutive and regulated alpha-secretase cleavage of Alzheimer's amyloid precursor protein by a disintegrin metalloprotease.",
        "authors": [
          "Lammich S.",
          "Kojro E.",
          "Postina R.",
          "Gilbert S.",
          "Pfeiffer R.",
          "Jasionowski M.",
          "Haass C.",
          "Fahrenholz F."
        ],
        "completeAuthorList": true,
        "publicationDate": "1999",
        "journal": "Proc. Natl. Acad. Sci. U.S.A.",
        "firstPage": "3922",
        "lastPage": "3927",
        "volume": "96",
        "literatureAbstract": "Amyloid beta peptide (Abeta), the principal proteinaceous component of amyloid plaques in brains of Alzheimer's disease patients, is derived by proteolytic cleavage of the amyloid precursor protein (APP). Proteolytic cleavage of APP by a putative alpha-secretase within the Abeta sequence precludes the formation of the amyloidogenic peptides and leads to the release of soluble APPsalpha into the medium. By overexpression of a disintegrin and metalloprotease (ADAM), classified as ADAM 10, in HEK 293 cells, basal and protein kinase C-stimulated alpha-secretase activity was increased severalfold. The proteolytically activated form of ADAM 10 was localized by cell surface biotinylation in the plasma membrane, but the majority of the proenzyme was found in the Golgi. These results support the view that APP is cleaved both at the cell surface and along the secretory pathway. Endogenous alpha-secretase activity was inhibited by a dominant negative form of ADAM 10 with a point mutation in the zinc binding site. Studies with purified ADAM 10 and Abeta fragments confirm the correct alpha-secretase cleavage site and demonstrate a dependence on the substrate's conformation. Our results provide evidence that ADAM 10 has alpha-secretase activity and many properties expected for the proteolytic processing of APP. Increases of its expression and activity might be beneficial for the treatment of Alzheimer's disease.",
        "statistics": {
          "reviewedProteinCount": 1,
          "unreviewedProteinCount": 0,
          "mappedProteinCount": 8
        }
      },
      "literatureMappedReference": {
        "uniprotAccession": "P05067",
        "source": "Reactome",
        "sourceId": "R-HSA-9010028",
        "annotation": "Pathway"
      },
      "categories": ["Function"],
      "publicationSource": "Computationally mapped"
    },
    {
      "id": "3",
      "uniProtReference": {
        "citation": {
          "citationType": "journal article",
          "authors": ["Borg J.-P.", "Ooi J.", "Levy E.", "Margolis B."],
          "citationXrefs": [
            {
              "databaseType": "PubMed",
              "id": "8887653"
            },
            {
              "databaseType": "DOI",
              "id": "10.1128/mcb.16.11.6229"
            }
          ],
          "title": "The phosphotyrosine interaction domains of X11 and FE65 bind to distinct sites on the YENPTY motif of amyloid precursor protein.",
          "publicationDate": "1996",
          "journal": "Mol. Cell. Biol.",
          "firstPage": "6229",
          "lastPage": "6241",
          "volume": "16"
        },
        "referencePositions": [
          "INTERACTION WITH APBA1 AND APBB1",
          "MUTAGENESIS OF TYR-728; TYR-757; ASN-759 AND TYR-762"
        ]
      },
      "categories": ["Interaction", "Pathol"],
      "publicationSource": "Swiss-Prot"
    }
  ]
}
