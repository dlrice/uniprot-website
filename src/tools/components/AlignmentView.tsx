import React, {
  useState,
  useEffect,
  useMemo,
  Dispatch,
  SetStateAction,
} from 'react';
import { DropdownButton } from 'franklin-sites';
import cn from 'classnames';

import Wrapped from './Wrapped';
import Overview from './Overview';

import {
  msaColorSchemeToString,
  MsaColorScheme,
} from '../config/msaColorSchemes';

import { getFullAlignmentLength } from '../utils/sequences';

import FeatureType from '../../uniprotkb/types/featureType';
import { FeatureData } from '../../uniprotkb/components/protein-data-views/FeaturesView';

import './styles/AlignmentView.scss';

export type ConservationOptions = {
  'calculate-conservation'?: true;
  'overlay-conservation'?: true;
};

export enum View {
  overview = 'Overview',
  wrapped = 'Wrapped',
}

export enum Tool {
  align = 'Align',
  blast = 'BLAST',
}

export type MSAInput = {
  name?: string;
  accession?: string;
  sequence: string;
  from: number;
  to: number;
  length: number;
  features?: FeatureData;
};

export type Sequence = {
  name: string;
  sequence: string;
  start: number;
  end: number;
  features?: FeatureData;
  accession?: string;
};

export type MSAViewProps = {
  alignment: MSAInput[];
  alignmentLength: number;
  highlightProperty: MsaColorScheme | undefined;
  conservationOptions: ConservationOptions;
  totalLength: number;
  annotation: FeatureType | undefined;
  activeId?: string;
  setActiveId: Dispatch<SetStateAction<string | undefined>>;
};

const AlignmentView: React.FC<{
  alignment: MSAInput[];
  alignmentLength: number;
  defaultView?: View;
  tool: Tool;
}> = ({ alignment, alignmentLength, defaultView, tool }) => {
  const annotationChoices = useMemo(() => {
    const features = alignment
      .map(({ features }) => features)
      .flat()
      .filter((features) => features);

    return Array.from(new Set(features?.map((feature) => feature?.type)));
  }, [alignment]);

  const [activeView, setActiveView] = useState<View>(
    defaultView || View.wrapped
  );
  const [annotation, setAnnotation] = useState<FeatureType | undefined>(
    annotationChoices[0]
  );
  const [highlightProperty, setHighlightProperty] = useState<MsaColorScheme>(
    MsaColorScheme.CLUSTAL
  );
  const [activeId, setActiveId] = useState<string | undefined>(
    alignment
      .filter(({ accession }) => accession)
      .map(({ accession }) => accession)[0]
  );

  const [checkedIds, setCheckedIds] = useState<{ [id: string]: boolean }>({});

  useEffect(() => {
    // if no default value was available on first render, set it now
    if (!annotation && annotationChoices.length) {
      setAnnotation(annotationChoices[0]);
    }
  }, [annotation, annotationChoices]);

  const totalLength = getFullAlignmentLength(alignment, alignmentLength);

  const conservationOptions: ConservationOptions =
    highlightProperty === MsaColorScheme.CONSERVATION
      ? {
          'calculate-conservation': true,
          'overlay-conservation': true,
        }
      : {};

  const handleSequenceChecked = (id: string) => {
    setCheckedIds({ ...checkedIds, [id]: !checkedIds[id] });
  };

  const AlignmentComponent = activeView === View.overview ? Overview : Wrapped;
  const additionalAlignProps =
    tool === Tool.align
      ? {
          setActiveId,
          omitInsertionsInCoords: true,
          onSequenceChecked: handleSequenceChecked,
        }
      : {};

  return (
    <>
      <div className="button-group">
        <DropdownButton label="Highlight properties" className="tertiary">
          <div className="dropdown-menu__content">
            <ul>
              {Object.entries(msaColorSchemeToString).map(
                ([schemeValue, schemeString]) => (
                  <li key={schemeString}>
                    <button
                      type="button"
                      className={cn('button', {
                        primary: highlightProperty === schemeValue,
                        tertiary: highlightProperty !== schemeValue,
                      })}
                      onClick={() =>
                        setHighlightProperty(schemeValue as MsaColorScheme)
                      }
                    >
                      {schemeString}
                    </button>
                  </li>
                )
              )}
            </ul>
          </div>
        </DropdownButton>
        {!!annotationChoices.length && (
          <DropdownButton label="Show annotation" className="tertiary">
            <div className="dropdown-menu__content">
              <ul>
                {annotationChoices.map((annotationChoice) => (
                  <li key={annotationChoice}>
                    <button
                      type="button"
                      className={cn('button', 'annotation-choice', {
                        primary: annotation === annotationChoice,
                        tertiary: annotation !== annotationChoice,
                      })}
                      onClick={() => setAnnotation(annotationChoice)}
                    >
                      {annotationChoice}
                    </button>
                  </li>
                ))}
              </ul>
            </div>
          </DropdownButton>
        )}
        <fieldset className="msa-view-choice">
          View:
          <label>
            <input
              name="view"
              type="radio"
              checked={activeView === View.overview}
              onChange={() => setActiveView(View.overview)}
            />
            Overview
          </label>
          <label>
            <input
              name="view"
              type="radio"
              checked={activeView === View.wrapped}
              onChange={() => setActiveView(View.wrapped)}
            />
            Wrapped
          </label>
        </fieldset>
      </div>
      <div>
        <AlignmentComponent
          alignment={alignment}
          alignmentLength={alignmentLength}
          highlightProperty={highlightProperty}
          conservationOptions={conservationOptions}
          totalLength={totalLength}
          annotation={annotation}
          activeId={activeId}
          {...additionalAlignProps}
        />
      </div>
    </>
  );
};

export default AlignmentView;
