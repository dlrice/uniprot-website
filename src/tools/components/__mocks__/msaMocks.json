{
  "BLAST": [
    {
      "name": "Query",
      "sequence": "MKRTVISSSNAYASKRSRLDIEHDFEQYHSLNKKYYPRPITRTGANQFNNKSRAKPMEIVEKLQKKQKTSFENVSTVMHWFRNDLRLYDNVGLYKSVALFQQLRQKNAKAKLYAVYVINEDDWRAHMDSGWKLMFIMGALKNLQQSLAELHIPLLLWEFHTPKSTLSNSKEFVEFFKEKCMNVSSGTGTIITANIEYQTDELYRDIRLLENEDHRLQLKYYHDSCIVAPGLITTDRGTNYSVFTPWYKKWVLYVNNYKKSTSEICHLHIIEPLKYNETFELKPFQYSLPDEFLQYIPKSKWCLPDVSEEAALSRLKDFLGTKSSKYNNEKDMLYLGGTSGLSVYITTGRISTRLIVNQAFQSCNGQIMSKALKDNSSTQNFIKEVAWRDFYRHCMCNWPYTSMGMPYRLDTLDIKWENNPVAFEKWCTGNTGIPIVDAIMRKLLYTGYINNRSRMITASFLSKNLLIDWRWGERWFMKHLIDGDSSSNVGGWGFCSSTGIDAQPYFRVFNMDIQAKKYDPQMIFVKQWVPELISSENKRPENYPKPLVDLKHSRERALKVYKDAM",
      "from": 1,
      "to": 565,
      "length": 565
    },
    {
      "name": "Match:P05066",
      "sequence": "MKRTVISSSNAYASKRSRLDIEHDFEQYHSLNKKYYPRPITRTGANQFNNKSRAKPMEIVEKLQKKQKTSFENVSTVMHWFRNDLRLYDNVGLYKSVALFQQLRQKNAKAKLYAVYVINEDDWRAHMDSGWKLMFIMGALKNLQQSLAELHIPLLLWEFHTPKSTLSNSKEFVEFFKEKCMNVSSGTGTIITANIEYQTDELYRDIRLLENEDHRLQLKYYHDSCIVAPGLITTDRGTNYSVFTPWYKKWVLYVNNYKKSTSEICHLHIIEPLKYNETFELKPFQYSLPDEFLQYIPKSKWCLPDVSEEAALSRLKDFLGTKSSKYNNEKDMLYLGGTSGLSVYITTGRISTRLIVNQAFQSCNGQIMSKALKDNSSTQNFIKEVAWRDFYRHCMCNWPYTSMGMPYRLDTLDIKWENNPVAFEKWCTGNTGIPIVDAIMRKLLYTGYINNRSRMITASFLSKNLLIDWRWGERWFMKHLIDGDSSSNVGGWGFCSSTGIDAQPYFRVFNMDIQAKKYDPQMIFVKQWVPELISSENKRPENYPKPLVDLKHSRERALKVYKDAM",
      "from": 1,
      "to": 565,
      "length": 565,
      "accession": "P05066",
      "features": [
        {
          "protvistaFeatureId": "cb1e3e80-227f-11eb-8c7f-bd249b5a2490",
          "start": 75,
          "end": 226,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "Other domain of interest",
          "description": "Photolyase/cryptochrome alpha/beta"
        },
        {
          "protvistaFeatureId": "cb1e3e81-227f-11eb-8c7f-bd249b5a2490",
          "start": 338,
          "end": 342,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "Nucleotide-binding region",
          "description": "FAD",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e82-227f-11eb-8c7f-bd249b5a2490",
          "start": 482,
          "end": 484,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "Nucleotide-binding region",
          "description": "FAD",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e83-227f-11eb-8c7f-bd249b5a2490",
          "start": 384,
          "end": 391,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "region of interest",
          "description": "Interaction with DNA",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e84-227f-11eb-8c7f-bd249b5a2490",
          "start": 451,
          "end": 452,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "region of interest",
          "description": "Interaction with DNA",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e85-227f-11eb-8c7f-bd249b5a2490",
          "start": 326,
          "end": 326,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "Other binding site",
          "description": "FAD",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e86-227f-11eb-8c7f-bd249b5a2490",
          "start": 514,
          "end": 514,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "Other binding site",
          "description": "DNA",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e87-227f-11eb-8c7f-bd249b5a2490",
          "start": 416,
          "end": 416,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "Other site of interest",
          "description": "Electron transfer via tryptophanyl radical",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e88-227f-11eb-8c7f-bd249b5a2490",
          "start": 469,
          "end": 469,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "Other site of interest",
          "description": "Electron transfer via tryptophanyl radical",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e89-227f-11eb-8c7f-bd249b5a2490",
          "start": 492,
          "end": 492,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "Other site of interest",
          "description": "Electron transfer via tryptophanyl radical",
          "evidences": [
            {
              "evidenceCode": "ECO:0000250"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e8a-227f-11eb-8c7f-bd249b5a2490",
          "start": 387,
          "end": 387,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "mutagenesis site",
          "description": "Reduces substrate binding 100-fold. Reduces quantum yield for dimer photolysis 3-fold.",
          "evidences": [
            {
              "evidenceCode": "ECO:0000269",
              "source": "PubMed",
              "id": "8344951"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e8b-227f-11eb-8c7f-bd249b5a2490",
          "start": 463,
          "end": 463,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "mutagenesis site",
          "description": "Reduces substrate binding 100-fold.",
          "evidences": [
            {
              "evidenceCode": "ECO:0000269",
              "source": "PubMed",
              "id": "8344951"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e8c-227f-11eb-8c7f-bd249b5a2490",
          "start": 507,
          "end": 507,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "mutagenesis site",
          "description": "Reduces substrate binding 100-fold.",
          "evidences": [
            {
              "evidenceCode": "ECO:0000269",
              "source": "PubMed",
              "id": "8344951"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e8d-227f-11eb-8c7f-bd249b5a2490",
          "start": 517,
          "end": 517,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "mutagenesis site",
          "description": "Reduces substrate binding 10-fold."
        },
        {
          "protvistaFeatureId": "cb1e3e8e-227f-11eb-8c7f-bd249b5a2490",
          "start": 77,
          "end": 77,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "sequence conflict",
          "description": "in Ref. 2; AAA34875",
          "evidences": [
            {
              "evidenceCode": "ECO:0000305"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e8f-227f-11eb-8c7f-bd249b5a2490",
          "start": 165,
          "end": 165,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "sequence conflict",
          "description": "in Ref. 2; AAA34875",
          "evidences": [
            {
              "evidenceCode": "ECO:0000305"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e90-227f-11eb-8c7f-bd249b5a2490",
          "start": 169,
          "end": 169,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "sequence conflict",
          "description": "in Ref. 2; AAA34875",
          "evidences": [
            {
              "evidenceCode": "ECO:0000305"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e91-227f-11eb-8c7f-bd249b5a2490",
          "start": 200,
          "end": 200,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "sequence conflict",
          "description": "in Ref. 2; AAA34875",
          "evidences": [
            {
              "evidenceCode": "ECO:0000305"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e92-227f-11eb-8c7f-bd249b5a2490",
          "start": 351,
          "end": 351,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "sequence conflict",
          "description": "in Ref. 2; AAA34875",
          "evidences": [
            {
              "evidenceCode": "ECO:0000305"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e93-227f-11eb-8c7f-bd249b5a2490",
          "start": 365,
          "end": 365,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "sequence conflict",
          "description": "in Ref. 2; AAA34875",
          "evidences": [
            {
              "evidenceCode": "ECO:0000305"
            }
          ]
        },
        {
          "protvistaFeatureId": "cb1e3e94-227f-11eb-8c7f-bd249b5a2490",
          "start": 473,
          "end": 473,
          "startModifier": "EXACT",
          "endModifier": "EXACT",
          "type": "sequence conflict",
          "description": "in Ref. 2; AAA34875",
          "evidences": [
            {
              "evidenceCode": "ECO:0000305"
            }
          ]
        }
      ]
    }
  ]
}
