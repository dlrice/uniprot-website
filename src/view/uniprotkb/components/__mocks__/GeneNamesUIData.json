{
  "name": "Gene name",
  "alternativeNames": ["name 2", "name 3", "name 4"]
}
