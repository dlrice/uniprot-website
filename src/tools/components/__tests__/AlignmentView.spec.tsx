import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderWithRedux from '../../../shared/__test-helpers__/RenderWithRedux';

import AlignmentView, { Tool, View } from '../AlignmentView';

import mockData from '../__mocks__/msaMocks.json';
import { resetUuidV1 } from '../../../../__mocks__/uuid';

describe('AlignmentView', () => {
  describe('BLAST', () => {
    let rendered;
    beforeEach(() => {
      resetUuidV1();
      const alignment = mockData.BLAST;
      rendered = renderWithRedux(
        <div className="main-content-and-footer">
          <AlignmentView
            alignment={alignment}
            alignmentLength={alignment[0].length}
            defaultView={View.overview}
            tool={Tool.blast}
          />
        </div>
      );
    });

    it('should render', () => {
      const { asFragment } = rendered;
      expect(asFragment()).toMatchSnapshot();
    });

    it('should update annotation on selection', () => {
      const { getByText, getByTestId } = rendered;
      const regionButton = getByText('Nucleotide-binding region');
      fireEvent.click(regionButton);
      const trackLabel = getByTestId('track-label');
      expect(trackLabel.textContent).toBe('Nucleotide-binding region');
    });

    it('should show toolip on feature click', () => {
      const { container, getByText } = rendered;
      //   const feature = container.getElementByClassname('feature-group');
      const track = container.querySelector('protvista-track');
      console.log(track);
      fireEvent.change(
        window,
        new CustomEvent('change', {
          detail: {
            eventtype: 'click',
            feature: {
              protvistaFeatureId: '7690a030-22a5-11eb-bbcb-e18b38a7b17f',
            },
            coords: [357, 933],
          },
        })
      );
      const tooltip = container.querySelector('protvista-tooltip');
      console.log(tooltip);
      console.log(tooltip.textContent);
      expect(tooltip).toBeTruthy();
      //   fireEvent.click(feature);
      //   expect(getByText('Other domain of interest 75-226')).toBeTruthy();
    });
  });
});
