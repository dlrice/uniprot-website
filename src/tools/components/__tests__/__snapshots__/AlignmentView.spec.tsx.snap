// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlignmentView BLAST should render 1`] = `
<DocumentFragment>
  <div
    class="main-content-and-footer"
  >
    <div
      class="button-group"
    >
      <div
        class="dropdown-container"
      >
        <button
          class="button dropdown tertiary"
          type="button"
        >
          Highlight properties
        </button>
        <div
          class="dropdown-menu"
        >
          <div
            class="autocomplete-container"
          >
            <span
              class="search-input"
            >
              <input
                data-testid="search-input"
                placeholder="Enter a color scheme"
                type="text"
                value=""
              />
              <span
                class="search-input__suffix"
                data-testid="search-input-suffix"
                role="presentation"
              >
                <svg
                  fill="currentColor"
                  height="14"
                  viewBox="0 0 512 512"
                  width="14"
                >
                  <path
                    d="M505 443L405 343c-4-4-10-7-17-7h-16a208 208 0 10-36 36v16c0 7 3 13 7 17l100 100c9 9 24 9 34 0l28-28c9-10 9-25 0-34zM208 336a128 128 0 110-256 128 128 0 010 256z"
                  />
                </svg>
              </span>
            </span>
            <div
              class="autocomplete-menu"
            >
              <div
                class="dropdown-menu__panel"
              />
            </div>
          </div>
          <div
            class="dropdown-menu__panel"
          >
            <ul
              class=""
            >
              <li
                class=""
              >
                <button
                  class="active"
                  type="button"
                >
                  Similarity
                </button>
              </li>
              <li
                class=""
              >
                <button
                  class=""
                  type="button"
                >
                  Clustal
                </button>
              </li>
              <li
                class="branch"
              >
                <button
                  class=""
                  type="button"
                >
                  Physical properties
                </button>
                <ul
                  class=""
                >
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Hydrophobicity
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Negative
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Positive
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Charged
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Polar
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Aliphatic
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Aromatic
                    </button>
                  </li>
                </ul>
              </li>
              <li
                class="branch"
              >
                <button
                  class=""
                  type="button"
                >
                  Structural properties
                </button>
                <ul
                  class=""
                >
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Buried Index
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Helix Propensity
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Strand Propensity
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Turn Propensity
                    </button>
                  </li>
                </ul>
              </li>
              <li
                class=""
              >
                <button
                  class=""
                  type="button"
                >
                  Serine Threonine
                </button>
              </li>
              <li
                class="branch"
              >
                <button
                  class=""
                  type="button"
                >
                  Other
                </button>
                <ul
                  class=""
                >
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Clustal 2
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Cinema
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Lesk
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      MAE
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Taylor
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Zappo
                    </button>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        class="dropdown-container"
      >
        <button
          class="button dropdown tertiary"
          type="button"
        >
          Select annotation
        </button>
        <div
          class="dropdown-menu"
        >
          <div
            class="autocomplete-container"
          >
            <span
              class="search-input"
            >
              <input
                data-testid="search-input"
                placeholder="Look for an annotation"
                type="text"
                value=""
              />
              <span
                class="search-input__suffix"
                data-testid="search-input-suffix"
                role="presentation"
              >
                <svg
                  fill="currentColor"
                  height="14"
                  viewBox="0 0 512 512"
                  width="14"
                >
                  <path
                    d="M505 443L405 343c-4-4-10-7-17-7h-16a208 208 0 10-36 36v16c0 7 3 13 7 17l100 100c9 9 24 9 34 0l28-28c9-10 9-25 0-34zM208 336a128 128 0 110-256 128 128 0 010 256z"
                  />
                </svg>
              </span>
            </span>
            <div
              class="autocomplete-menu"
            >
              <div
                class="dropdown-menu__panel"
              />
            </div>
          </div>
          <div
            class="dropdown-menu__panel"
          >
            <ul
              class=""
            >
              <li
                class="branch"
              >
                <button
                  class="active"
                  type="button"
                >
                  Match:P05066
                </button>
                <ul
                  class=""
                >
                  <li
                    class=""
                  >
                    <button
                      class="active"
                      type="button"
                    >
                      Other domain of interest
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Nucleotide-binding region
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      region of interest
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Other binding site
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      Other site of interest
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      mutagenesis site
                    </button>
                  </li>
                  <li
                    class=""
                  >
                    <button
                      class=""
                      type="button"
                    >
                      sequence conflict
                    </button>
                  </li>
                </ul>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <fieldset
        class="msa-view-choice"
      >
        View:
        <label>
          <input
            checked=""
            name="view"
            type="radio"
          />
          Overview
        </label>
        <label>
          <input
            name="view"
            type="radio"
          />
          Wrapped
        </label>
      </fieldset>
    </div>
    <div>
      <protvista-tooltip />
      <section
        class="alignment-grid"
        data-testid="alignment-view"
      >
        <span
          class="track-label"
        >
          Overview
        </span>
        <div
          class="track"
        >
          <div>
            <protvista-track
              height="15"
              highlight="1:566"
              layout="non-overlapping"
              length="565"
            />
            <protvista-track
              height="15"
              highlight="1:566"
              layout="non-overlapping"
              length="565"
            />
          </div>
        </div>
        <span
          class="track-label"
          data-testid="track-label"
        >
          Other domain of interest
        </span>
        <div
          class="track"
        >
          <protvista-track
            highlight="1:566"
            layout="non-overlapping"
            length="565"
          />
        </div>
        <div
          class="track-label track-label--align-labels"
        >
          <span
            class="align-label"
          >
            Query
          </span>
          <button
            class="align-label"
            style="height: 20px;"
            tabindex="-1"
            type="button"
          >
            Match:
            <a
              href="/uniprotkb/P05066"
            >
              P05066
            </a>
          </button>
        </div>
        <div
          class="track"
        >
          <protvista-manager
            attributes="displaystart displayend"
            displayend="565"
            displaystart="1"
          >
            <protvista-navigation
              length="565"
            />
            <protvista-msa
              calculate-conservation="true"
              colorscheme="conservation"
              height="40"
              hidelabel="true"
              length="565"
              overlay-conservation="true"
            />
          </protvista-manager>
        </div>
        <span
          class="right-coord"
        >
          <div
            style="height: 20px;"
          >
            565
          </div>
          <div
            style="height: 20px;"
          >
            565
          </div>
        </span>
      </section>
    </div>
  </div>
</DocumentFragment>
`;
